### ISSUE TITLE
[Monitor Query] Improve workspace ID troubleshooting

### ISSUE THREAD

**Issue by pvaneck** [Azure/azure-sdk-for-python#32798]
There have some cases where customers have been trying to use `LogsQueryClient.query_workspace` by passing in a resource ID of a workspace resource instead of the workspace ID.  Investigation and perhaps some validation should be done for cases like this.

Some options:
- Add validation, checking that workspace ID doesn't begin with `subscription/*` indicating that resource ID was passed in. Suggest using a workspace ID in this case. 
- Improve the docstring with an example workspace ID, so users can see the expected format.
- Improve error message when an invalid/nonexistent workspace ID is passed to the service. Currently, the message returned is `The requested path does not exist`. This can be improved.

**Comment by twacn** [comment]
I agree. It is very confusing. For example, the `LogAnalyticsManagementClient.workspaces.list` returns a Workspace object where the `id` is actually the resource ID -- the "Workspace ID" is actually called `customer_id`.
